prime-router/src/main/kotlin/history/SubmissionHistory.kt

```diff
@@ -287,12 +288,21 @@ class DetailedSubmissionHistory(
      */
     fun enrichWithDescendants(descendants: List<DetailedSubmissionHistory>) {
         check(descendants.distinctBy { it.actionId }.size == descendants.size)
+
         // Enforce an order on the enrichment:  process/translate, send, download
-        descendants.filter {
-            it.actionName == TaskAction.process || it.actionName == TaskAction.translate
-        }.forEach { descendant ->
-            enrichWithProcessAction(descendant)
+        if (topic == "full-elr") {
+            // logs and destinations are handled very differently for UP
+            // both routing and translate are populated at different times,
+            // so we need to do special logic to handle them
+            enrichWithRoutingAndTranslationActions(descendants)
```

Comment:
If this is specific to UP, is it worth including the name of the pipeline here? Is it worth it?

Reply:
Thinking of a good name....anyone who sees this, feel free to suggest ideas

Reply:
My naive opinion here would be that we don't need it in the name (the name is pretty long to begin with). The comment here already calls out that it is specific to UP, and the function comments for `enrichWithRoutingAndTranslationActions` and `enrichWithProcessAction` already mention that they are specific to one or the other pipeline.

Reply:
Will leave as is for now